### PR TITLE
[sui-json] Support Vec<StructInstantiation>

### PR DESCRIPTION
**Pull request by icodezjb**
Support `vector` parameters with generic type

companion: #4836

### CODE REVIEW

**Comment by sblackshear** [comment]
@awelc @tnowacki: I think we need this in order to support (e.g.) `entry fun f<T>(coins: vector<Coin<T>>)`. Please check my understanding on this

File: crates/sui-json/src/lib.rs [hunk 1]
```diff
@@ -486,18 +486,19 @@ fn resolve_call_arg(
             idx,
             &arg.to_json_value(),
         )?)),
-        SignatureToken::Vector(inner) => {
-            if let SignatureToken::Struct(_) = &**inner {
+        SignatureToken::Vector(inner) => match &**inner {
+            SignatureToken::Struct(_) | SignatureToken::StructInstantiation(_, _) => {
```
Comment (tnowacki) [inline crates/sui-json/src/lib.rs:490]: Ah good catch!

Comment (tnowacki) [reply crates/sui-json/src/lib.rs:490]: cc @awelc

**Comment by tnowacki** [comment]
> @awelc @tnowacki: I think we need this in order to support (e.g.) `entry fun f<T>(coins: vector<Coin<T>>)`. Please check my understanding on this

Agreed, I think it is necessary (at least from the SDK side of things)

File: crates/sui-core/src/unit_tests/data/entry_point_vector/sources/objects_vector.move [hunk 2]
```diff
@@ -111,6 +123,91 @@ module entry_point_vector::entry_point_vector {
...
+    public entry fun mint_shared_any<Any>(v: u64, ctx: &mut TxContext) {
+        transfer::share_object(
+            ObjAny<Any> {
+                id: object::new(ctx),
+                value: v,
+            }
+        )
+    }
+
+    public entry fun obj_vec_destroy_any<Any>(v: vector<ObjAny<Any>>, _: &mut TxContext) {
```
Comment (sblackshear) [inline crates/sui-core/src/unit_tests/data/entry_point_vector/sources/objects_vector.move:165]:
```suggestion
    public entry fun obj_vec_destroy_any<Any>(v: vector<ObjAny<Any>>) {
```

Here and elsewhere, unused `TxContext` is not needed anymore!

File: crates/sui-json/src/lib.rs (hunk 1 above)
Comment (awelc) [reply crates/sui-json/src/lib.rs:490]: Makes sense to me.

**Review by awelc — COMMENTED** [review]
LGTM

**Review by awelc — APPROVED** [review]
LGTM

**Pull request merged**